src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx

```diff
@@ -354,9 +350,19 @@ export default class ThirdPartyCollectionDetailPage extends React.PureComponent<
     const hasAccess = this.hasAccess()
     const shouldRender = hasAccess && collection
     return (
-      <LoggedInDetailPage className="ThirdPartyCollectionDetailPage" hasNavigation={!hasAccess && !isLoading} isLoading={isLoading}>
-        {shouldRender && thirdParty ? this.renderPage(thirdParty) : <NotFound />}
-      </LoggedInDetailPage>
+      <CollectionProvider id={collection?.id}>
+        {({ isLoading: isLoadingCollectionData }) => {
+          return (
+            <LoggedInDetailPage
+              className="ThirdPartyCollectionDetailPage"
+              hasNavigation={!hasAccess && !isLoading && !isLoadingCollectionData}
+              isLoading={isLoading || isLoadingCollectionData}
+            >
+              {shouldRender && thirdParty ? this.renderPage(thirdParty) : <NotFound />}
+            </LoggedInDetailPage>
+          )
+        }}
+      </CollectionProvider>
```

Comment:
Question!
I think we are leaving the `XXXProvider`s after the `LoggedInDetailPage` so we don't fetch the resource for an address that doesn't have the permissions to see it. In this case I can see us fetching the collection before knowing if the address is connected or not.

So! why are we loading here and not on `renderPage`'s level